bokehjs/src/coffee/renderer/tile/mercator_tile_source.coffee

```diff
@@ -0,0 +1,262 @@
+_ = require "underscore"
+TileSource = require "./tile_source"
+
+class MercatorTileSource extends TileSource
+
+  type: 'MercatorTileSource'
+  
+  defaults: =>
+    return _.extend {}, super(), {
+      tile_size : 256
+      max_zoom : 25
+      min_zoom : 0
+      extra_url_vars : {}
+      full_extent : [-20037508.34, -20037508.34, 20037508.34, 20037508.34]
+      x_origin_offset : 20037508.34
+      y_origin_offset : 20037508.34
+      initial_resolution : 2 * Math.PI * 6378137 / 256
+      resolutions : [
+        156543.03392804097
+        78271.51696402048
+        39135.75848201024
+        19567.87924100512
+        9783.93962050256
+        4891.96981025128
+        2445.98490512564
+        1222.99245256282
+        611.49622628141
+        305.748113140705
+        152.8740565703525
+        76.43702828517625
+        38.21851414258813
+        19.109257071294063
+        9.554628535647032
+        4.777314267823516
+        2.388657133911758
+        1.194328566955879
+        0.5971642834779395
+        0.29858214173896974
+        0.14929107086948487
+        0.07464553543474244
+        0.03732276771737122
+        0.01866138385868561
+        0.009330691929342804
```

Comment:
Ah, I guess they are all derived as powers of two with initial_resolution as a base...